### PR TITLE
Add new method returns ThreadInfo

### PR DESCRIPTION
**Pull request by jpbempel**
Requires by SamplerWriter

### CODE REVIEW

File: dd-trace-core/src/main/java/datadog/trace/core/util/JmxThreadStackProvider.java [hunk 1]
```diff
@@ -30,4 +26,24 @@
     }
     return stackTraces;
   }
+
+  @Override
+  public List<ThreadInfo> getThreadInfo(List<Long> threadIds) {
```
Comment (richardstartin) [inline dd-trace-core/src/main/java/datadog/trace/core/util/JmxThreadStackProvider.java:31]: Where does the need for this to be a collection originate from? Can we make do with this being a `long[]` everywhere given that this won't be public API (won't it?)

File: dd-trace-core/src/main/java/datadog/trace/core/util/NoneThreadStackProvider.java [hunk 2]
```diff
@@ -9,4 +10,9 @@
   public List<StackTraceElement[]> getStackTrace(List<Long> threadIds) {
     return Collections.emptyList();
   }
+
+  @Override
+  public List<ThreadInfo> getThreadInfo(List<Long> threadIds) {
```
Comment (richardstartin) [inline dd-trace-core/src/main/java/datadog/trace/core/util/NoneThreadStackProvider.java:15]: Who consumes this API? Is there a reason they can't consume `ThreadInfo[]` given that's what we're wrapping elsewhere?

**Comment by richardstartin** [comment]
It's small fry but can we avoid some of the conversions? Is `long[]` good enough to represent a list of thread ids and is `ThreadInfo[]` good enough to represent a list of `ThreadInfo`?

File: dd-trace-core/src/main/java/datadog/trace/core/util/JmxThreadStackProvider.java (hunk 1 above)
Comment (jpbempel) [reply dd-trace-core/src/main/java/datadog/trace/core/util/JmxThreadStackProvider.java:31]: The underlying call is indeed a long[] for the JMX call, but in the end we may change for other thing. So I won't to lock the api to an implementation.

File: dd-trace-core/src/main/java/datadog/trace/core/util/NoneThreadStackProvider.java (hunk 2 above)
Comment (jpbempel) [reply dd-trace-core/src/main/java/datadog/trace/core/util/NoneThreadStackProvider.java:15]: same remark than above

File: dd-trace-core/src/main/java/datadog/trace/core/util/JmxThreadStackProvider.java (hunk 1 above)
Comment (richardstartin) [reply dd-trace-core/src/main/java/datadog/trace/core/util/JmxThreadStackProvider.java:31]: Does the other thing take a `List<Long>` and return a `List<ThreadInfo>`? The wrapping here doesn't really _abstract_ the underlying call.

**Pull request merged**